sites/geohub/src/lib/types/Layer.ts

```diff
@@ -9,4 +10,6 @@ export interface Layer {
   children?: Layer[]
   parentId?: string
   dataset: StacItemFeature
+  colormapName?: string
+  classificationMethod?: ClassificationMethodTypes
```

Comment:
You need to change to use this property instead of using map.fire.

But you maybe can do this change in another PR.

Reply:
I plan to remove those fire events later, once the refactoring is done